### PR TITLE
Web interface for iris module latest version(2.5.2)

### PR DESCRIPTION
**Pull request by Moksh45**
Fixes #

#### Resolves
add web interface for the iris module

#### Checklist

- [x] I have read the Contribution & Best practices Guideline.
- [x] My branch is up-to-date with the Upstream `master` branch.
- [x] The acceptance, integration, unit tests pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by saeeddhqan** [comment]
is it okay to have this error:
sh: 1: react-scripts: not found
What is the endpoint for the iris?

**Comment by Moksh45** [comment]
react-scripts: not found," typically occurs when the react-scripts package is missing or not installed correctly. This package is required for running and building React applications.
1. Please make sure that you have Node.js and npm (Node Package Manager) installed on your system.
2. After running the command "pip install -e ." to set up Maryam, please navigate to the path "Maryam/core/web/web_interface" and verify the existence of the "node_modules" directory.

and When I try to run the web interface, no errors appear in my terminal.

Screenshot from 2023-06-23 10-55-43

**Comment by Moksh45** [comment]
What is the endpoint for the iris? I'm not sure what you're asking.

**Comment by saeeddhqan** [comment]
If Node.js is a requirement, please verify that it exists before running the server. If it does not exist, print a message.

**Comment by Moksh45** [comment]
ok

**Comment by Moksh45** [comment]
I have completed all the necessary changes. Please take a moment to review the Pull Request again and verify the modifications. If you have any further feedback or suggestions, let me know. Thank you

**Comment by saeeddhqan** [comment]
Remove this section:
image
Remove this section:
image
This section is terrible:
image
Put the logo on the right side. And crop the logo like this:
image
The query should remain in the form. The search form has a pink square. Remove it.
Use this color for the background: #000d02
And the black color for the header section: #000
Make the search form precisely like startpage.com.
Remove icons from this section:
image
Add something like this to the footer, left side:
image
The color is #174d07.
Change the title to Iris. Not Iris web interface.
Please correct the home form too:
image
Change the home bc color to: #000d02.
The home owasp logo is too big. Make it smaller and crop it as I said above.

**Comment by Moksh45** [comment]
Ok I made All the changes

File: maryam/core/initial.py [hunk 1]
```diff
@@ -156,6 +157,39 @@ def do_web(self, params):
...
 		if 'api' not in sys.modules:
 			from .web import api
+   
+		if params[0] == 'iris':
+			# Check if npm is installed
+			npm_exists = shutil.which('npm') is not None
+			# Check if node is installed
+			node_exists = shutil.which('node') is not None
+   
+			if not npm_exists or not node_exists:
```
Comment (saeeddhqan) [inline maryam/core/initial.py]: change it to: if not (npm_exists and node_exists)

File: maryam/core/initial.py [hunk 2]
```diff
@@ -156,6 +157,39 @@ def do_web(self, params):
...
+				print("npm and/or Node.js are not installed.")
```
Comment (saeeddhqan) [inline maryam/core/initial.py]: do not use double-quotes for simple-form strings.

**Comment by Moksh45** [comment]
ss
In this particular section, I am retaining the code in the code base but making use of the \<Search hideShortCut /> feature.

the result looks like 
Screenshot from 2023-06-23 15-32-16

**Comment by saeeddhqan** [comment]
We do not need them. Not even in the future.

**Comment by Moksh45** [comment]
ok, i will remove that specific part from the code base.

**Comment by Moksh45** [comment]
I have created a design for the search header field that I believe matches your requirements. If you approve the design, I will proceed with implementing it.
WhatsApp Image 2023-06-24 at 1 37 01 AM

**Comment by saeeddhqan** [comment]
Put the logo at the far left side.

**Comment by Moksh45** [comment]
like this!
WhatsApp Image 2023-06-24 at 12 46 25 PM

**Comment by saeeddhqan** [comment]
Fine.

**Comment by Moksh45** [comment]
I believe I have resolved all the errors in the UI and made the necessary updates to the web interface based on your suggestions. The final result looks like this. 

Screenshot from 2023-06-24 23-12-42
image


#### I kindly request you review my pull request once again

File: maryam/core/initial.py (hunk 2 above)
Comment (saeeddhqan) [reply maryam/core/initial.py]: resolve this

**Comment by saeeddhqan** [comment]
Much better.

1. Now, when I click on the search form, it shows a pink border. Resolve it. Make it like startpage.com, go to the website and when you click on the form, it shows a light purple color.
2. Add an under construction section to the image tab or hide it.

**Comment by Moksh45** [comment]
The search form now has a border added, and the image tab has been hidden on the search result page. Please review the pull request

**Comment by saeeddhqan** [comment]
I still see the pink border.

**Comment by saeeddhqan** [comment]
Check this error:
image

When I go to the http://localhost:3000/search without sending a request for the backend:

Screenshot from 2023-06-28 11-03-37
You need to redirect them to the index page. It's like going to the https://www.startpage.com/sp/search without a query.

**Comment by Moksh45** [comment]
Alright, I'll change the border color of the search box to a light purple. Additionally, I'll fix the existing error. Lastly, I'll add the functionality to redirect users to the index page.

**Comment by Moksh45** [comment]
Apologies for accidentally clicking on the "Close PR"

**Comment by saeeddhqan** [comment]
Remove the pink rectangle please.

**Comment by Moksh45** [comment]
I don't understand which pink rectangle you're referring to. Could you please share a picture or image of the pink rectangle with me?

**Comment by Moksh45** [comment]
I have made all the necessary adjustments based on your suggestions. Please review the changes accordingly

ss

#### I believe I have resolved this error, but could you please confirm if you encountered the same issue?

**Comment by saeeddhqan** [comment]
I still see the pink rectangle. And it does not show anything. I tried iris and it works but the web interface does not show anything.

**Comment by Moksh45** [comment]
I have resolved the border issue. please check it

**Pull request closed without merging**